CrustTests/CRMapperTests.swift

```diff
@@ -21,7 +21,8 @@ class MockMap: Mapping, Adapter {
         catchMapping!(toMap, context)
     }
     
-    func mappingBegins() throws { }
+    var mappingDidBegin: Bool = false
+    func mappingBegins() throws { self.mappingDidBegin = true }
```

Comment:
`isCurrentlyMapping` might be too verbose, but could work. `isMapping` sounds like it's checking a class type. 

For whatever reason, I don't like the fact that one is WillBegin and the other one is DidEnd. It feels asymmetrical. That said, it probably makes the most sense in this context, so 👍